ganga/GangaGUI/gui/routes.py

```diff
@@ -1,14 +1,81 @@
+import jwt
 from GangaGUI.gui import app
-from flask import request
+from flask import request, jsonify
+from functools import wraps
+from GangaGUI.gui.models import User
 
 
 @app.route("/")
 def dashboard():
     return "Hello GangaGUI"
 
 
-@app.route('/shutdown', methods=['POST'])
+# Generate token for API authentication - token validity 5 days
```

Comment:
What happens if the user runs the Ganga session and GUI for more than 5 days? Do they need to stop and start the GUI again?

Reply:
Nope, simply renew the token. If the user is just using the GUI, then no worries I'll add one popup just re-enter the password and token will be renewed.